src/shared/watchdog.c

```diff
@@ -96,6 +97,10 @@ static int open_watchdog(void) {
         return update_timeout();
 }
 
+void watchdog_set_path(const char *path) {
+	watchdog_dev = path;
```

Comment:
as discussed above, this should do strdup(). And hence also be able to return an error...

And I figure this should be called watchdog_set_device() or so